### PR TITLE
Start implementing semantics for a Web3 RPC server

### PR DESCRIPTION
**Pull request by gtrepta**

### CODE REVIEW

File: web3-json-rpc.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    configuration
+      <kevm />
+      <web3request>
+        <jsonrpc> "" </jsonrpc>
+        <callid> 0 </callid>
+        <method> "" </method>
+        <params> [ .JSONList ] </params>
+      </web3request>
+      <web3result> { .JSONList } </web3result>
```
Comment (ehildenb) [inline web3-json-rpc.md]: Let's have a wrapper configuration:

```
configuration
  <kevm-client>
    <kevm/>
    ...
  </kevm-client>
```

File: web3-json-rpc.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    syntax EthereumSimulation ::= JSON
+    syntax JSON ::= Int | Bool
+
+    rule <k> J:JSON => #loadRPCCall J ... </k>
+
+    syntax KItem ::= "#loadRPCCall" JSON
+ // ----------------------------------
```
Comment (ehildenb) [inline web3-json-rpc.md]: formatting

File: web3-json-rpc.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    syntax KItem ::= "#runRPCCall"
+ // ------------------------------
+    rule <k> #runRPCCall => . </k>
+         <method> "net_version" </method>
+         <web3result> _ => #runNetVersion </web3result>
+
+    syntax JSON ::= "#runNetVersion"
+ // ---------------------------------
+    rule <web3result> #runNetVersion => { "id" : CALLID, "jsonrpc" : JSONRPC, "result" : Int2String( CHAINID ) } </web3result>
```
Comment (ehildenb) [inline web3-json-rpc.md]: <k> #runNetVersion => #sendResponse { ... } ... </k>

Comment (ehildenb) [reply web3-json-rpc.md]: `<k> #sendResponse RESPONSE => .  ... </k> <response> ... (.List => ListItem(RESPONSE)) </response>`

File: web3.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+         <activeAccounts> ACCTS </activeAccounts>
+
+    syntax JSONList ::= #acctsToJArray( Set ) [function]
+ // ------------------------------------------------
+    rule #acctsToJArray( .Set ) => .JSONList
+    rule #acctsToJArray( SetItem( ACCT ) ACCTS:Set ) => Int2String( ACCT ), #acctsToJArray( ACCTS )
+
+    syntax KItem ::= "#eth_getBalance"
+ // ----------------------------------
+    rule <k> #eth_getBalance => #sendResponse( { "id" : CALLID, "jsonrpc" : JSONRPC, "result" : Int2String( ACCTBALANCE ) } ) ... </k>
```
Comment (ehildenb) [inline web3.md]: Can convert to hex string with `String2Base(...)` from `domains.k`.

**Comment by ehildenb** [comment]
Make a new method `#unparseQuantity : Int -> String` and `#unparseData : ByteArray -> String` which encode the values we need to pass back correctly.

File: web3.md [hunk 5]
```diff
@@ -125,5 +129,34 @@ module WEB3
            <balance> ACCTBALANCE </balance>
          ... </account>
 
+    syntax KItem ::= "#eth_getStorageAt"
+ // -------------------------------
+    rule <k> #eth_getStorageAt ... </k>
+         <params> [ (DATA => #parseHexWord(DATA)), QUANTITY:Int, _ ] </params>
```
Comment (gtrepta) [inline web3.md]: QUANTITY should be a hex string. We will have to revisit `eth_getStorageAt`, as its parameters can involve keccak encodings. https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getstorageat

Comment (dwightguth) [reply web3.md]: the work involving keccak actually happens on the other side, we shouldn't need to worry about it.

Comment (gtrepta) [reply web3.md]: But we would have to decode it when it comes in, or I am looking at this the wrong way?

**Pull request merged**